slickml/optimization.py

```diff
@@ -335,3 +339,158 @@ def plot_optimization_results(self):
         Function to plot the optimization results.
         """
         pass
+
+
+class XGBoostClassifierHyperOpt(XGBoostClassifier):
+    """XGBoost Classifier - Hpyerparameter Optimization.
+    This class uses HyperOpt, a Python library for serial and parallel
+    optimization over search spaces, which may include real-valued, discrete,
+    and conditional dimensions to train a XGBoost model.Main reference is
+    HyperOpt GitHub: (https://github.com/hyperopt/hyperopt)
+    Parameters
+    ----------
+    num_boost_round: int, optional (default=200)
+        Number of boosting round at each fold of xgboost.cv()
+    metrics: str or tuple[str], optional (default=("auc"))
+        Metric used for evaluation at cross-validation
+        using xgboost.cv(). Please note that this is different
+        than eval_metric that needs to be passed to params dict.
+        Possible values are "auc", "aucpr", "error", "logloss"
+    n_splits: int, optional (default=4)
+        Number of folds for cross-validation
+    shuffle: bool, optional (default=True)
+        Flag to shuffle data to have the ability of building
+    early_stopping_rounds: int, optional (default=20)
+        The criterion to early abort the xgboost.cv() phase
+        if the test metric is not improved
+        early_stopping_rounds: int, optional (default=20)
+    verbost: str, optional (default=False)
+        Print evaluation results from model
+
+    Attributes
+    ----------
+    process: class method
+        Performs hyperparameter tunning on input dataset
+    xgb_cv: class method
+        Optimization function for XGBoost utilizing cross-validation
+    """
+
+    def __init__(
+        self,
+        params=None,
+        num_boost_rounds=200,
```

Comment:
These all should None and re-define in __init__. Either they should inherit or re-instantiated